src/nccl_ofi_topo.c

```diff
@@ -241,7 +241,9 @@ static void enable_hwloc_io_types(hwloc_topology_t topo)
 	hwloc_topology_set_io_types_filter(topo, filter);
 #else
 	unsigned long flags = hwloc_topology_get_flags(topo);
-	flags |= HWLOC_TOPOLOGY_FLAG_IO_DEVICES | HWLOC_TOPOLOGY_FLAG_IO_BRIDGES;
+	flags |= HWLOC_TOPOLOGY_FLAG_IO_DEVICES		/* Common I/O devices - NICs, GPUs, etc */
+		 | HWLOC_TOPOLOGY_FLAG_IO_BRIDGES	/* Bridges with at least 1 device */
+		 | HWLOC_TOPOLOGY_FLAG_WHOLE_IO;	/* Generic/uncommon devices and bridges without devices */
```

Comment:
My reading of https://www.open-mpi.org/projects/hwloc/doc/v1.11.13/a00124.php#ggada025d3ec20b4b420f8038d23d6e7bdea4a41dc181649ef81c2dcd44a54e271b9 is that you don't need the DEVICES and BRIDGES flags if you're specifying the WHOLE_IO flag.

Reply:
That' correct, debated that a bit and left it in there anyway. I can clean it up.